service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadDocumentsAboutToSubmitControllerTest.java

```diff
@@ -0,0 +1,100 @@
+import uk.gov.hmcts.reform.fpl.model.common.Document;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.ApplicationDocumentsService;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static uk.gov.hmcts.reform.fpl.enums.YesNo.YES;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.callbackRequest;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(UploadDocumentsController.class)
+@OverrideAutoConfiguration(enabled = true)
+public class UploadDocumentsAboutToSubmitControllerTest extends AbstractControllerTest {
+    @MockBean
+    private ApplicationDocumentsService applicationDocumentsService;
+
+    @Autowired
+    private static Time time;
+
+
+    UploadDocumentsAboutToSubmitControllerTest() {
+        super("upload-documents");
+    }
+
+    private static final String USER = "kurt@swansea.gov.uk";
+    private static final CaseDetails caseDetails = callbackRequest().getCaseDetails();
+    private static final CaseDetails caseDetailsBefore = callbackRequest().getCaseDetailsBefore();
+    private static final LocalDate date = LocalDate.of(2021,10, 13);
+
+    @BeforeEach
+    void init() {
+        Map<String, Object> updatedCaseData = new HashMap<>();
+
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        List<Element<ApplicationDocument>> updatedDocuments = caseData.getDocuments();
+        updatedDocuments.get(0).getValue().setUploadedBy(USER);
+        updatedDocuments.get(0).getValue().setDateTimeUploaded(LocalDateTime.of(date,
+            LocalTime.of(13, 30)));
+        updatedCaseData.put("documents", updatedDocuments);
+
+        CaseData caseDataBefore = mapper.convertValue(caseDetailsBefore.getData(), CaseData.class);
+
+        given(applicationDocumentsService.updateCaseDocuments(caseData.getDocuments(), caseDataBefore.getDocuments()))
+            .willReturn(updatedCaseData);
+    }
+
+    @Test
+    void shouldSetShowCreatedByAndDateTimeUploadedFlagToYes() {
+        CaseDetails caseDetails = callbackRequest().getCaseDetails();
+
+        CaseDetails caseDetailsBefore = callbackRequest().getCaseDetailsBefore();
+
+        CallbackRequest callbackRequest = CallbackRequest.builder()
+            .caseDetails(caseDetails)
+            .caseDetailsBefore(caseDetailsBefore)
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postAboutToSubmitEvent(callbackRequest);
+
+        assertThat(callbackResponse.getData().get("showCreatedByAndDateTimeUploadedFlag").equals(YES));
+    }
+
+    @Test
+    void shouldUpdateDocumentsInCaseDataWithCreatedByAndDateTimeUploadedFieldsIncluded() {
```

Comment:
Do we need another test here just to make sure that old document properties are persisted?

Reply:
I have now extended this test to check that the existing fields are present also!